src/components/Layout/BurgerMenuContent.tsx

```diff
@@ -1,42 +1,124 @@
-import React, {useContext, useState, useEffect} from 'react'
+import React, { useContext, useState, useEffect } from 'react'
 import { BurgerMenuStatusContext } from '../Context/BurgerMenuStatusContext'
+import SearchInputMobile from '../Search/SearchInputMobile'
+import CategoryList from '../Category/CategoryList'
+import StatusFilter from '../Search/StatusFilter'
+import TimeFrameFilter from '../Search/TimeFrameFilter'
+import MobileNavigation from './MobileNavigation'
+import CategoryFilter from '../Search/CategoryFilter'
 import './BurgerMenuContent.css'
 
-type BurgerMenuContentProps = {
-  footerTranslate: string,
-  children: React.ReactNode
+export type FilterStatus = {
+  categoryOpen: boolean
+  statusOpen: boolean
+  timeFrameOpen: boolean
 }
 
-function BurgerMenuContent({footerTranslate, children} : BurgerMenuContentProps) {
+export type BurgerMenuContentProps = {
+  navigationOnly?: boolean
+}
+
+const filtersInitialStatus = { categoryOpen: true, statusOpen: false, timeFrameOpen: false }
+
+const OPEN_BURGER_HEIGHT = 584
+const SEARCH_TITLE_HEIGHT = 32
+const CATEGORY_FILTER_HEIGHT = 368
+const STATUS_FILTER_HEIGHT = 290
+const TIMEFRAME_FILTER_HEIGHT = 212
+const CLOSED_FILTER_HEIGHT = 56
+const MOBILE_NAVIGATION_HEIGHT = 104
+
+function BurgerMenuContent({ navigationOnly }: BurgerMenuContentProps) {
   const [footer, setFooter] = useState<Element | null>(null)
   const burgerMenu = useContext(BurgerMenuStatusContext)
+  const [filterStatus, setFilterStatus] = useState(filtersInitialStatus)
+  const [burgerOpen, burgerSearching, burgerFiltering, burgerTranslate] = [
```

Comment:
You can do destructuring here `burgerMenu?.status` to avoid creating a new array
`const { burgerOpen, burgerSearching, burgerFiltering, burgerTranslate } = burgerMenu?.status`

Reply:
+1